src/server/rdb_load.cc

```diff
@@ -505,9 +512,10 @@ void RdbLoaderBase::OpaqueObjLoader::CreateSet(const LoadTrace* ltrace) {
       return true;
     });
   } else {
-    bool use_set2 = GetFlag(FLAGS_use_set2);
+    bool use_set2 = GetFlag(FLAGS_use_set2) || rdb_type_ == RDB_TYPE_SET_WITH_EXPIRY;
```

Comment:
maybe we should print an error use_set2 is false and rdb_type_ is RDB_TYPE_SET_WITH_EXPIRY